snp_motion_planning/src/plugins/tasks/tcp_speed_limiter_task.cpp

```diff
@@ -0,0 +1,158 @@
+
+  bool operator==(const TCPSpeedLimiterTask& rhs) const
+  {
+    bool equal = true;
+    equal &= tesseract_planning::TaskComposerTask::operator==(rhs);
+    return equal;
+  }
+  bool operator!=(const TCPSpeedLimiterTask& rhs) const
+  {
+    return !operator==(rhs);
+  }
+
+protected:
+  friend struct tesseract_common::Serialization;
+  friend class boost::serialization::access;
+  tesseract_planning::TaskComposerNodeInfo::UPtr runImpl(tesseract_planning::TaskComposerContext& context,
+                                                         OptionalTaskComposerExecutor /*executor*/) const override final
+  {
+    auto info = std::make_unique<tesseract_planning::TaskComposerNodeInfo>(*this);
+    info->return_value = 0;
+
+    // Get the problem
+    auto& problem = dynamic_cast<tesseract_planning::PlanningTaskComposerProblem&>(*context.problem);
+
+    // --------------------
+    // Check that inputs are valid
+    // --------------------
+    auto input_data_poly = context.data_storage->getData(input_keys_[0]);
+    if (input_data_poly.isNull() ||
+        input_data_poly.getType() != std::type_index(typeid(tesseract_planning::CompositeInstruction)))
+    {
+      info->message = "Input results to TCP speed limiter task must be a composite instruction";
+      CONSOLE_BRIDGE_logError("%s", info->message.c_str());
+      return info;
+    }
+
+    auto& ci = input_data_poly.as<tesseract_planning::CompositeInstruction>();
+    const tesseract_common::ManipulatorInfo& manip_info = ci.getManipulatorInfo();
+
+    // Get Composite Profile
+    std::string profile = ci.getProfile();
+    profile = tesseract_planning::getProfileString(name_, profile, problem.composite_profile_remapping);
+    auto cur_composite_profile = tesseract_planning::getProfile<TCPSpeedLimiterProfile>(
+        name_, profile, *problem.profiles, std::make_shared<TCPSpeedLimiterProfile>());
+    cur_composite_profile =
+        tesseract_planning::applyProfileOverrides(name_, profile, cur_composite_profile, ci.getProfileOverrides());
+
+    // Create data structures for checking for plan profile overrides
+    auto flattened = ci.flatten(tesseract_planning::moveFilter);
+    if (flattened.empty())
+    {
+      info->message = "TCP speed limiter task found no MoveInstructions to process";
+      info->return_value = 1;
+      return info;
+    }
+
+    // Solve using parameters
+    tesseract_planning::TrajectoryContainer::Ptr trajectory =
+        std::make_shared<tesseract_planning::InstructionsTrajectory>(ci);
+    std::vector<std::string> joint_names = problem.env->getJointGroup(manip_info.manipulator)->getJointNames();
+
+    TCPSpeedLimiter solver(problem.env);
+
+    if (!solver.compute(*trajectory, joint_names, cur_composite_profile->max_speed, cur_composite_profile->tcp))
+    {
+      info->message = "Failed to perform TCP speed limiter task for process input: " + ci.getDescription();
+      return info;
+    }
```

Comment:
Call the function instead, and use try-catch to log any errors

```suggestion
    try
    {
      limitTCPSpeed(...);
    }
    catch(const std::exception& ex)
    {
      info->message = ex.what();
      return info;
    }
```